truffleruby/src/main/ruby/core/process_mirror.rb

```diff
@@ -166,28 +164,10 @@ def initialize(env_or_cmd, *args)
             if should_use_shell?(@command)
               @command, @argv = '/bin/sh', ['sh', '-c', @command]
             else
-              # If the command contains both the binary to run and the arguments, we need to split them apart. We have
-              # two basic cases here: 1) a fully qualified command; and 2) a simple name expected to be found on the PATH.
-              # Both cases require the split. In the event of a fully qualified command, we exec the command directly,
-              # but the signature for exec requires the command and arguments to all be split. In the other case, where
-              # we have a command to search on the PATH, we must split the command apart from the arguments in order to
-              # perform the search (a poor man's version of shell processing). If we can find it on the PATH, then we
-              # run the whole thing through a shell to get proper shell processing.
-
+              # No shell processing, and posix_spawnp will do the lookup of the command if needed
               split_command, *split_args = @command.strip.split(' ')
-
-              if should_search_path?(split_command)
-                resolved_command = resolve_in_path(split_command)
-
-                if resolved_command
-                  @command, @argv = '/bin/sh', ['sh', '-c', @command]
-                else
-                  raise Errno::ENOENT.new("No such file or directory - #{@command}")
-                end
-              else
-                @command = split_command
-                @argv = [split_command] + split_args
-              end
+              @command = split_command
+              @argv = [split_command] + split_args
             end
           else
             # If arguments are explicitly passed, the semantics of this method (defined in Ruby) are to run the
```

Comment:
I think we can also get rid of the manual PATH lookup below, `posix_spawnp` can take of it better, and it allows to give a pid in case the lookup fails.